src/OrchardCore.Modules/OrchardCore.Queries/Recipes/QueryStep.cs

```diff
@@ -42,22 +43,42 @@ public async Task ExecuteAsync(RecipeExecutionContext context)
             }
 
             var model = context.Step.ToObject<QueryStepModel>(_jsonSerializerOptions);
-
+            var queries = new List<Query>();
             foreach (var token in model.Queries.Cast<JsonObject>())
             {
+                var name = token[nameof(Query.Name)].ToString();
+
+                if (string.IsNullOrEmpty(name))
+                {
+                    _logger.LogError("Query name is missing or empty. The query will not be imported.");
```

Comment:
Both of the logged errors here are better suited to be displayed to the user via `RecipeExecutionContext.Errors`.

Reply:
yes. great idea,